modules/spicetify/spicetify.nix

```diff
@@ -0,0 +1,56 @@
+{ config, options, lib, pkgs, ... }:
+
+let
+
+  colors = with config.lib.stylix.colors; ''
+    [base]
+    text               = ${base05}
+    subtext            = ${base05}
+    main               = ${base00}
+    main-elevated      = ${base02}
+    highlight          = ${base02}
+    highlight-elevated = ${base03}
+    sidebar            = ${base01}
+    player             = ${base05}
+    card               = ${base04}
+    shadow             = ${base00}
+    selected-row       = ${base05}
+    button             = ${base05}
+    button-active      = ${base05}
+    button-disabled    = ${base04}
+    tab-active         = ${base02}
+    notification       = ${base02}
+    notification-error = ${base08}
+    equalizer          = ${base0B}
+    misc               = ${base02}
+  '';
+
+in {
+  options.stylix.targets.spicetify.enable =
+    config.lib.stylix.mkEnableTarget "Spicetify" true;
+
+  config = lib.mkIf (config.stylix.enable && config.stylix.targets.spicetify.enable && (config.programs?spicetify)) (
+    lib.optionalAttrs (builtins.hasAttr "spicetify" options.programs) {
+      programs.spicetify = {
+        theme = {
+          name = "stylix";
+          src = pkgs.stdenv.mkDerivation {
+            name = "stylix-spicetify-theme";
+            unpackPhase = "true"; # Don't load any source
+            buildPhase = ''
+              cat << EOF > ./color.ini
+              ${colors}
+              EOF
+            '';
+            installPhase = ''
+              mkdir -p $out
+              cp ./color.ini $out
+            '';
```

Comment:
Could this be simplified using `pkgs.writeTextDir`?

Reply:
Oh I was hoping there were functions like these when I asked in the NixOS discord server, but got no response 😆. I will look into it!

Reply:
I ended up using `pkgs.writeTextDir`. `builtins.toFile` doesn't seem to create a directory with a file of the given name, which is requred here.